ci/ctest_to_gitlab.sh

```diff
@@ -98,20 +98,28 @@ JOB_TEMPLATE="
     USE_MPI: 'YES'
     DISABLE_AFTER_SCRIPT: 'YES'
     DLAF_HDF5_TEST_OUTPUT_PATH: \$CI_PROJECT_DIR
-  script: mpi-ctest -L {{LABEL}}"
+  script: mpi-ctest -L {{CATEGORY_LABEL}} -L {{RANK_LABEL}}"
 fi
 
 JOBS=""
 
-for label in `ctest --print-labels | egrep -o "RANK_[1-9][0-9]?"`; do
-    N=`echo "$label" | sed "s/RANK_//"`
-    C=$(( THREADS_PER_NODE / N ))
+for rank_label in `ctest --print-labels | egrep -o "RANK_[1-9][0-9]?"`; do
+    for category_label in `ctest --print-labels | egrep -o "CATEGORY_[A-Z]+"`; do
+        N=`echo "$rank_label" | sed "s/RANK_//"`
+        C=$(( THREADS_PER_NODE / N ))
 
-    JOB=`echo "$JOB_TEMPLATE" | sed "s|{{LABEL}}|$label|g" \
-                              | sed "s|{{NTASKS}}|$N|g" \
-                              | sed "s|{{CPUS_PER_TASK}}|$C|g"`
+        # Skip label combinations that match no tests
+        if ! ctest -N -L $category_label -L $rank_label | egrep --quiet "^Total Tests: [1-9][0-9]?$"; then
+            continue
+        fi
 
-    JOBS="$JOBS$JOB"
+        JOB=`echo "$JOB_TEMPLATE" | sed "s|{{CATEGORY_LABEL}}|$category_label|g" \
```

Comment:
Bikeshedding: what about dropping the `CATEGORY_` prefix, to make the job name a bit less verbose and more readable? Probably personal preference, but I prefer `MINIAPP_RANK_6` to `CATEGORY_MINIAPP_RANK_6` (i.e. the `CATEGORY` part is a bit redundant IMO, since it will be added to all tests).

Reply:
Yep, that sounds reasonable I think. I initially just wanted to keep it as straightforward of a mapping from CTest labels to CI job names, but I'd be ok with dropping the prefix there.

Just checking: are you suggesting to drop the `CATEGORY_` prefix from the job name only, or also from the CTest label? For the latter it's still handy to be explicit; `CATEGORY` and `RANK` essentially act as keys in key-value pairs. The filtering is a bit more straightforward I think if `CATEGORY` stays in the label name.

Reply:
> are you suggesting to drop the `CATEGORY_` prefix from the job name only, or also from the CTest label?

Initially I thought it would be nice to drop it from both, but then I realized you were parsing the label as you point out. That's why I suggested to drop it only from CI, just for having a slightly less bloated output in the GitLab UI.

Reply:
Thoughts from the others?

Reply:
https://github.com/eth-cscs/DLA-Future/pull/1112/commits/dfb60874b642cc05bf34f549e7898e1c6f7cd0b1